### PR TITLE
add support for running the event loop

### PR DESCRIPTION
**Pull request by avi**
allow running of promise/async code

You need to poll runEventloop manually for this to happen allowing the embedder full control

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,6 +23,8 @@ CFLAGS_EMCC+=-s INVOKE_RUN=0
 CFLAGS_EMCC+=-s ALLOW_MEMORY_GROWTH=1
 CFLAGS_EMCC+=-s ALLOW_TABLE_GROWTH=1
 CFLAGS_EMCC+=-s FILESYSTEM=0
+CFLAGS_EMCC+=-s RESERVED_FUNCTION_POINTERS=1024
+CFLAGS_EMCC+=-s BINARYEN_TRAP_MODE='clamp'
```
Comment (justjake) [inline Makefile:27]: why is this added?

Comment (justjake) [reply Makefile:27]: update: as of #15, this project officially uses emcc 1.39.18 - I don't think BINARYEN_TRAP_MODE is supported on the newer emcc version. emscripten setup is now documented in README.md, and the build is tested in CI.

Comment (avi) [reply Makefile]: I had problems loading a large source file without it - it might have been my older emcc version.
I'll open a separate PR on this if the issue persists
Reverted both changes

**Comment by avi** [comment]
From an ergonomics viewpoint I wonder if it is better to model it as a successOrFail or make it a function that returns a number on success (number of executed jobs) and throws on error.
SuccessOrFail is more consistent with the rest of the codebase but is slightly harder to use and will probably cause people to silently drop errors sometimes...
I left it as you suggested for now
